src/__Private/PHPWarningSuppressor.hack

```diff
@@ -0,0 +1,32 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST\__Private;
+
+/**
+ * Many PHP builtins emit warnings to stderr when they fail. This
+ * class allows us to squash warnings for a time without using PHP's
+ * `@` annotation.
+ */
+final class PHPWarningSuppressor implements \IDisposable {
```

Comment:
In general... _Private is private, but it's probably better to use `\HH\Lib\_Private\PHPWarningSuppressor` than copy-paste.

Perhaps we should move that to `HH\Lib\PHP` or something like that.

Reply:
I prefer not using the HSL private version. Publishing it under a non private name has my preference. I use this code via copy and paste in my own projects. For now, I have made hhast use the `_Private` hsl version and made a PR on the hsl itself to publish the warning suppressor.